markets/bfp-market/contracts/modules/PerpAccountModule.sol

```diff
@@ -124,6 +141,130 @@ contract PerpAccountModule is IPerpAccountModule {
+        PerpMarketConfiguration.Data storage marketConfig = PerpMarketConfiguration.load(marketId);
+        Position.Data storage fromPosition = market.positions[fromId];
+
+        // From account should not be liquidatable
+        if (
+            Position.isLiquidatable(
+                fromPosition,
+                market,
+                runtime.oraclePrice,
+                marketConfig,
+                Margin.getMarginUsd(fromId, market, runtime.oraclePrice)
+            )
+        ) {
+            revert ErrorUtil.CanLiquidatePosition();
+        }
+
+        // Move collaterals from `from` -> `to`.
+        Margin.GlobalData storage globalMarginConfig = Margin.load();
+        Margin.Data storage fromAccountMargin = Margin.load(fromId, marketId);
+        PerpMarketConfiguration.GlobalData storage globalConfig = PerpMarketConfiguration.load();
+
+        runtime.supportedSynthMarketIdsLength = globalMarginConfig.supportedSynthMarketIds.length;
+
+        for (uint256 i = 0; i < runtime.supportedSynthMarketIdsLength; ) {
+            runtime.synthMarketId = globalMarginConfig.supportedSynthMarketIds[i];
+            runtime.fromAccountCollateral = fromAccountMargin.collaterals[runtime.synthMarketId];
+            if (runtime.fromAccountCollateral > 0) {
+                // Move collateral `from` -> `to`.
+                runtime.collateralToMove = runtime.fromAccountCollateral.mulDecimal(proportion);
+                toAccountMargin.collaterals[runtime.synthMarketId] = runtime.collateralToMove;
+                fromAccountMargin.collaterals[runtime.synthMarketId] -= runtime.collateralToMove;
+                runtime.toCollateralUsd += runtime.collateralToMove.mulDecimal(
+                    globalMarginConfig.getCollateralPrice(runtime.synthMarketId, globalConfig)
+                );
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+
+        // Move debt from `from` -> `to`.
+        runtime.debtToMove = fromAccountMargin.debtUsd.mulDecimal(proportion).to128();
+        toAccountMargin.debtUsd = runtime.debtToMove;
+        fromAccountMargin.debtUsd -= runtime.debtToMove;
+
+        // Move position from `from` -> `to`.
+        runtime.sizeToMove = fromPosition.size.mulDecimal(proportion.toInt()).to128();
+
+        if (fromPosition.size < 0) {
+            fromPosition.size += MathUtil.abs(runtime.sizeToMove).toInt().to128();
+        } else {
+            fromPosition.size -= runtime.sizeToMove;
+        }
+
+        toPosition.update(
+            Position.Data(
+                runtime.sizeToMove,
+                fromPosition.entryTime,
+                fromPosition.entryFundingAccrued,
+                fromPosition.entryUtilizationAccrued,
+                fromPosition.entryPrice
+            )
+        );
+
+        // Make sure the `toAccount` has enough margin for IM.
+        (runtime.im, , ) = Position.getLiquidationMarginUsd(
```

Comment:
Can we check if also above mm?

Reply:
If it passes IM it will always pass MM?

Reply:
Ah yep you're right. `maintenanceMarginScalar` will be < 1 and MM always < IM.